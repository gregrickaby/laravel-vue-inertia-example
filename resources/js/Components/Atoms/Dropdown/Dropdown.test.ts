import DropDown from '@/Components/Atoms/DropDown/DropDown.vue'
import { mount } from '@vue/test-utils'

describe('DropDown', () => {
    /**
     * Mount the component.
     */
    const wrapper = mount(DropDown, {
        slots: {
            trigger: '<button>Trigger</button>',
            content: '<div>Content</div>'
        }
    })

    /**
     * Assert the component renders.
     */
    test('renders', () => {
        expect(wrapper.exists()).toBeTruthy()
    })

    /**
     * Assert the dropdown opens and closes correctly.
     */
    test('opens and closes', async () => {
        // Cast the element to HTMLElement to access the style property.
        const dropdownElement = wrapper.find('.absolute').element as HTMLElement

        // Initially, the dropdown should be closed.
        expect(
            dropdownElement.style.display === 'none' ||
                dropdownElement.style.display === ''
        ).toBe(true)

        // Simulate a click to open the dropdown.
        await wrapper.find('button').trigger('click')
        await wrapper.vm.$nextTick()
        expect(dropdownElement.style.display).not.toBe('none')

        // Simulate another click to close the dropdown.
        await wrapper.find('.fixed.inset-0').trigger('click')
        await wrapper.vm.$nextTick()
        expect(
            dropdownElement.style.display === 'none' ||
                dropdownElement.style.display === ''
        ).toBe(true)
    })

    /**
     * Assert the component matches the snapshot.
     */
    test('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
