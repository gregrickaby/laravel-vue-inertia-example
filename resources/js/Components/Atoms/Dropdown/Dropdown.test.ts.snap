// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropDown > matches snapshot 1`] = `
"<div class="relative">
  <div><button>Trigger</button></div><!-- Full Screen Dropdown Overlay -->
  <div class="fixed inset-0 z-40" style="display: none;"></div>
  <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 scale-95" entertoclass="opacity-100 scale-100" leaveactiveclass="transition ease-in duration-75" leavefromclass="opacity-100 scale-100" leavetoclass="opacity-0 scale-95" appear="false" persisted="true" css="true">
    <div class="absolute z-50 mt-2 rounded-md shadow-lg w-48 ltr:origin-top-right rtl:origin-top-left end-0" style="display: none;">
      <div class="rounded-md ring-1 ring-black ring-opacity-5 py-1 bg-white dark:bg-gray-700">
        <div>Content</div>
      </div>
    </div>
  </transition-stub>
</div>"
`;
